File: apps/desktop/src-tauri/src/models/capture.rs
use serde::{Deserialize, Serialize};

/// Information required to launch or confirm a capture workflow.
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct CaptureContext {
    #[serde(default)]
    pub session_id: Option<String>,
    #[serde(default)]
    pub record_id: Option<String>,
    #[serde(default)]
    pub asset_id: Option<String>,
    #[serde(default)]
    pub target_seconds: Option<i64>,
    #[serde(default)]
    pub actual_seconds: Option<f64>,
    #[serde(default)]
    pub result_save_path: Option<String>,
}

pub type CaptureOverlayPayload = CaptureContext;

/// Logical monitor bounds reported by the renderer when preparing a capture.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CaptureMonitor {
    pub x: i32,
    pub y: i32,
    pub width: u32,
    pub height: u32,
    pub scale_factor: f64,
}

/// Rectangle describing the selection region to capture.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CaptureRect {
    pub x: i32,
    pub y: i32,
    pub width: u32,
    pub height: u32,
}

/// Payload provided when requesting a capture preview.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CapturePreviewPayload {
    pub rect: CaptureRect,
    pub monitor: CaptureMonitor,
}

/// Data returned to the renderer for preview display.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CapturePreview {
    pub base_url: String,
}

File: apps/desktop/src/shared/types/capture.ts
export interface CaptureRect {
  x: number;
  y: number;
  width: number;
  height: number;
}

// Bounds are expressed in logical screen points, matching browser event coordinates.
export interface CaptureMonitor {
  x: number;
  y: number;
  width: number;
  height: number;
  scaleFactor: number;
}

export interface CapturePreviewPayload {
  rect: CaptureRect;
  monitor: CaptureMonitor;
}

export interface CapturePreview {
  baseUrl: string;
}

export interface CaptureContext {
  sessionId?: string | null;
  recordId?: string | null;
  assetId?: string | null;
  targetSeconds?: number | null;
  actualSeconds?: number | null;
  resultSavePath?: string | null;
}

export type CaptureOverlayPayload = CaptureContext;
